### DISCUSSION TITLE
Deactivate "trailingSlash" logic

### DISCUSSION THREAD

**Discussion by vikingair** [vercel/next.js#39719] [Ideas]
### Describe the feature you'd like to request

Currently (according to [this](https://nextjs.org/docs/api-reference/next.config.js/trailing-slash)) it is only possible to decide wether NextJS should or should not use trailing slashes. Using a middleware on top of that doing further redirects can result into more than a single redirect, which is bad in terms of SEO. It would be nice if these internal trailing-slash redirects could be deactivated and handled manually in the middleware to allow single redirects. 

### Describe the solution you'd like

```js
module.exports = {
  // we still need the config to now for client links and redirects if we want to use the trailing slash
  // but setting "noServerRedirect" to "true" we want to disable the server logic that applies before running the
  // middleware code
  trailingSlash: { preferred: true, noServerRedirect: true },
}
```

### Describe alternatives you've considered

We've considered forwarding sanitized URLs from Virtual-Service to the NextJS application, but ran into issues because NextJS server didn't like this mismatch and wouldn't allow to send the correct headers for a redirect back. We seem to have no control over these internal redirects. Allowing us to deactivate these, gives us more control over the incoming requests.

**Comment by SukkaW** [comment]
> ...doing further redirects can result into more than a single redirect, which is bad in terms of SEO

That is not true. For Google, Bing, and Yandex, as long as every member of redirects' chain all have the desired status code, this will not impact SEO.

https://developers.google.com/search/docs/advanced/crawling/301-redirects

**Comment by neruson** [comment]
I would also be interested in this, but for another reason. I'm brand new to Next.js and learning by attempting to deploy a basic static page to Firebase. However, because of the trailing slash redirect, Firebase can't deploy without a Cloud Functions backend, which isn't available on a free account. ([firebase-tools source](https://github.com/firebase/firebase-tools/blob/master/src/frameworks/next/index.ts#L174-L177))

Given that Firebase is a competing platform, and a static site is not the usual Next.js use case, I could see this not being a priority, but I think it's a stumbling block that could prevent new users from exploring Next.js more- it's certainly dampened my enthusiasm a bit to not be able to deploy a simple Hello World page.

**Reply by neruson** [reply]
Never mind, I just learned about the new `skipTrailingSlashRedirect` option in version 13.1! It might be nice to add to the [Trailing Slash](https://nextjs.org/docs/api-reference/next.config.js/trailing-slash) section of the docs and not just [Advanced Middleware Flags](https://nextjs.org/docs/advanced-features/middleware#advanced-middleware-flags), but I'm happy to see it exists.

**Comment by amit-911** [comment]
But ,I have issue with Next.js 14.2.24 to trailingSlash should set true. google search engine show all routes are folder than links
